entity-service-client/src/main/java/org/hypertrace/entity/data/service/client/EdsCacheClient.java

```diff
@@ -17,26 +17,35 @@
 import org.hypertrace.entity.data.service.v1.EntityRelationship;
 import org.hypertrace.entity.data.service.v1.EntityRelationships;
 import org.hypertrace.entity.data.service.v1.Query;
+import org.hypertrace.entity.service.client.config.EntityServiceClientCacheConfig;
 import org.hypertrace.entity.service.client.config.EntityServiceClientConfig;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class EdsCacheClient implements EdsClient {
 
   private static final Logger LOG = LoggerFactory.getLogger(EdsCacheClient.class);
-  private final LoadingCache<EdsCacheKey, Optional<EnrichedEntity>> enrichedEntityCache;
-  private final LoadingCache<EdsCacheKey, Optional<Entity>> entityCache;
+  private LoadingCache<EdsCacheKey, Optional<EnrichedEntity>> enrichedEntityCache;
+  private LoadingCache<EdsCacheKey, Optional<Entity>> entityCache;
+  private LoadingCache<EdsTypeAndIdAttributesCacheKey, Optional<String>> entityIdsCache;
 
   private final EntityDataServiceClient client;
 
   EdsCacheClient(EntityServiceClientConfig entityServiceClientConfig) {
     this(new EntityDataServiceClient(entityServiceClientConfig));
+    initCache(entityServiceClientConfig.getCacheConfig());
```

Comment:
This will initialize the caches twice because the `this()` call would have initialized with DEFAULT values already.
How about passing the cacheConfig as second param to the other constructor.

Reply:
Hmm. The other one is also public interface `public EdsCacheClient(EntityDataServiceClient client)`, and there was one consumer that directly uses this. So, I had to override this way.

Reply:
You can either add one more constructor or change the public constructor too. Consumer will change in next version upgrade.

Reply:
Done.